src/Components/Radio/Radio.component.jsx

```diff
@@ -108,6 +124,21 @@ const Radio = ({activeTrack, setActiveTrack}) => {
     }
   }
 
+  useEffect( 
+    () => {
+      if (playing) {
+        //update on play, then every 10 seconds
+        fetchData();
+        const interval = setInterval(
+          () => {
+            fetchData(); 
+          }, 10000);
+        //cleanup effect
+        return () => clearInterval(interval);
```

Comment:
blocker 

- this runs like "componentDidUnmount", notice that you are returning a function that calls a function not calling a function. That function gets called at the end of the work day, given that the radio does not unmount until the app is closed this interval is not going to quite get the job done, but the clean up is good

-  We need to be killing the interval when the user clicks pause or eject

Reply:
since this interval only exists in the scope of useEffect, why even cleanup? i copied this cleanup from a stackoverflow post. more so, how do i cleanup that interval if i can't access it from outside. i could declare it outside, and add as dependency. then cleanup on pause/eject. i'll see how it runs

Reply:
The interval does not only exist in the scope of useEffect. When you call setInterval you are offloading async work into the Timer module built into the Node environment.  The interval is going to run in node then get loaded  into the microtask queue, then once the call stack in the main process is idle the event loop will take the first item from the mircotask queue, which will be the function you included in the setInterval call